verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -257,6 +268,34 @@ VName KytheFactsExtractor::ExtractModuleInstanceFact(
   return module_instance_vname;
 }
 
+VName KytheFactsExtractor::ExtractModuleNamedPort(
+    const IndexingFactNode& named_port_node) {
+  const auto& port_name = named_port_node.Value().Anchors()[0];
+
+  // TODO(minatoma): Change this to use the general scope that will be created.
+  //
+  // Parent Node must be kModuleInstance and the grand parent node must be
+  // kDataTypeReference.
+  const Anchor& module_type =
+      named_port_node.Parent()->Parent()->Value().Anchors()[0];
+  const VName* named_port_module_vname =
+      scope_context_.SearchForDefinition(CreateSignature(module_type.Value()));
+
+  const VName actual_port_vname(
+      file_path_, CreateScopeRelativeSignature(
+                      port_name.Value(), named_port_module_vname->signature));
+  const VName port_vname_anchor = PrintAnchorVName(port_name);
+  GenerateEdgeString(port_vname_anchor, kEdgeRef, actual_port_vname);
+
+  if (named_port_node.Children().empty()) {
+    const VName* definition_vname =
+        scope_context_.SearchForDefinition(CreateSignature(port_name.Value()));
```

Comment:
What if definition is not found?

Reply:
I wrapped it inside `ABSL_DIE_IF_NULL` for now but this will solved in a follow PR. We have some issues about the crashing.

Reply:
ok for now